src/Main.elm

```diff
@@ -1,10 +1,198 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser exposing (sandbox)
+import Data.Audience exposing (..)
+import Data.AudienceFolder exposing (..)
+import Dict exposing (Dict)
+import Dict.Extra as DE exposing (fromListDedupe)
+import Html exposing (Html, div, button, text)
+import Html.Events
+import Html.Attributes
+import Json.Decode as JD exposing (decodeString)
+import List as L exposing (map, drop, filter)
+import Maybe as M
 
 
-main : Html msg
-main =
-    Html.text "There will be app soon!"
+type alias Model = {
+    audiences : Dict Int (List Audience)
+    , audienceFolders :  Dict Int (List AudienceFolder)
+    , currentFolder : List Int
+    , audienceFilter : AudienceType }
+
+type Msg =
+    Up
+    | Down Int
```

Comment:
Not much descriptive to be honest. In this case it's fine, but from "this will be maintained by various people in future" point of view I would rather choose something more clear for the first look.
Like `OpenFolder AudienceFolderId` and `BackFromFolder` or anything else. `Int` also did not telling much about it's real type (folderId).

Reply:
Yes, you are right. The action names should have been more descriptive.